cadc-util/src/main/java/ca/nrc/cadc/auth/BearerTokenPrincipal.java

```diff
@@ -0,0 +1,115 @@
+/*
+ ************************************************************************
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ *
+ * (c) 2019.                         (c) 2019.
+ * National Research Council            Conseil national de recherches
+ * Ottawa, Canada, K1A 0R6              Ottawa, Canada, K1A 0R6
+ * All rights reserved                  Tous droits reserves
+ *
+ * NRC disclaims any warranties         Le CNRC denie toute garantie
+ * expressed, implied, or statu-        enoncee, implicite ou legale,
+ * tory, of any kind with respect       de quelque nature que se soit,
+ * to the software, including           concernant le logiciel, y com-
+ * without limitation any war-          pris sans restriction toute
+ * ranty of merchantability or          garantie de valeur marchande
+ * fitness for a particular pur-        ou de pertinence pour un usage
+ * pose.  NRC shall not be liable       particulier.  Le CNRC ne
+ * in any event for any damages,        pourra en aucun cas etre tenu
+ * whether direct or indirect,          responsable de tout dommage,
+ * special or general, consequen-       direct ou indirect, particul-
+ * tial or incidental, arising          ier ou general, accessoire ou
+ * from the use of the software.        fortuit, resultant de l'utili-
+ *                                      sation du logiciel.
+ *
+ *
+ * @author cbanek
+ * 8/05/19 - 5:01 PM
+ *
+ *
+ *
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ ************************************************************************
+ */
+package ca.nrc.cadc.auth;
+
+
+import ca.nrc.cadc.util.StringUtil;
+
+import java.io.Serializable;
+import java.security.Principal;
+
+
+/**
+ * Represents a bearer token
+ */
+public class BearerTokenPrincipal implements Principal, Serializable
+{
+    private static final long serialVersionUID = 7L;
+    private static final String prefix = "Bearer ";
+
+    private final String token;
+
+
+    public BearerTokenPrincipal(final String authorizationHeader)
+    {
+        if (!isBearerToken(authorizationHeader))
+        {
+            throw new IllegalArgumentException("Not a bearer token");
+        }
+
+        this.token = authorizationHeader.substring(prefix.length());
+    }
+
+    public static Boolean isBearerToken(final String authorizationHeader)
+    {
+        return StringUtil.hasText(authorizationHeader) &&
+               authorizationHeader.startsWith(prefix);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "BearerTokenPrincipal[" + getToken().substring(0, 8) + "]";
+    }
+
+    /**
+     * Returns the name of this principal.
+     *
+     * @return the name of this principal.
+     */
+    @Override
+    public String getName()
+    {
+        return getToken();
+    }
+
+    public String getToken()
```

Comment:
I probably would not bother with getToken() and just make the javadoc for getName() clear that it returns the entire token.